lib/prefix.c

```diff
@@ -1088,14 +1088,48 @@ void prefix_mcast_inet4_dump(const char *onfail, struct in_addr addr,
 	errno = save_errno;
 }
 
+void prefix_mcast_inet6_dump(const char *onfail, struct in6_addr addr,
+			     char *buf, int buf_size)
+{
+	int save_errno = errno;
+
+	if (memcmp(&addr, &in6addr_any, sizeof(struct in6_addr)) == 0)
+		strlcpy(buf, "*", buf_size);
+	else {
+		if (!inet_ntop(AF_INET6, &addr, buf, buf_size)) {
+			if (onfail)
+				snprintf(buf, buf_size, "%s", onfail);
+		}
+	}
+
+	errno = save_errno;
+}
+
 const char *prefix_sg2str(const struct prefix_sg *sg, char *sg_str)
 {
-	char src_str[INET_ADDRSTRLEN];
-	char grp_str[INET_ADDRSTRLEN];
+	if (sg->family == AF_INET) {
+		char src_str[INET_ADDRSTRLEN];
+		char grp_str[INET_ADDRSTRLEN];
 
-	prefix_mcast_inet4_dump("<src?>", sg->src, src_str, sizeof(src_str));
-	prefix_mcast_inet4_dump("<grp?>", sg->grp, grp_str, sizeof(grp_str));
-	snprintf(sg_str, PREFIX_SG_STR_LEN, "(%s,%s)", src_str, grp_str);
+		prefix_mcast_inet4_dump("<src?>", sg->src.ipaddr_v4, src_str,
+					sizeof(src_str));
+		prefix_mcast_inet4_dump("<grp?>", sg->grp.ipaddr_v4, grp_str,
+					sizeof(grp_str));
+
+		snprintf(sg_str, PREFIX_SG_STR_LEN, "(%s,%s)", src_str,
+			 grp_str);
+	} else if (sg->family == AF_INET6) {
+		char src_str[INET6_ADDRSTRLEN];
+		char grp_str[INET6_ADDRSTRLEN];
+
+		prefix_mcast_inet6_dump("<src?>", sg->src.ipaddr_v6, src_str,
+					sizeof(src_str));
+		prefix_mcast_inet6_dump("<grp?>", sg->grp.ipaddr_v6, grp_str,
+					sizeof(grp_str));
+
+		snprintf(sg_str, PREFIX_SG_STR_LEN, "(%s,%s)", src_str,
```

Comment:
Let's increase the PREFIX_SG_STR_LEN as 82 to accommodate IPv6 (S,G) with null character at the end. 
Max possible ==> (dead:beef:dead:beef:dead:beef:dead:beef,dead:beef:dead:beef:dead:beef:dead:beef)
(39 *2) [S,G] + 3 [for (,)] + 1 [null char] ==> 82

Reply:
Thanks Rathna for this comment. This is a Work in progress PR. We have kept it in draft. The changes are still in progress. Once the changes are complete wrt data structures, we will mark it as "ready for review". We are working on it.